homework/index.js

```diff
@@ -29,26 +34,128 @@
     return elem;
   }
 
+  function addRepoDetailsInfo(listElm, title, value) {
+    const trElm = createAndAppend('tr', listElm, { class: 'tr-elm' });
+    createAndAppend('td', trElm, {
+      class: 'titles',
+      text: title,
+    });
+
+    createAndAppend('td', trElm, {
+      class: 'values',
+      text: value !== null ? value : 'No Information',
+    });
+
+    return trElm;
+  }
+
+  function formatDate(stringDateAndTime) {
+    const dt = new Date(stringDateAndTime);
+    return dt.toLocaleString();
+  }
+  function sortReposByNameProp(firstRepo, secondRepo) {
+    return firstRepo.name.localeCompare(secondRepo.name);
+  }
+
   function renderRepoDetails(repo, ul) {
-    createAndAppend('li', ul, { text: repo.name });
+    const li = createAndAppend('li', ul, { class: 'repos' });
+    const table = createAndAppend('table', li);
+    const firstRow = addRepoDetailsInfo(table, 'Repository :', '');
+    createAndAppend('a', firstRow.lastChild, {
+      text: repo.name,
+      href: repo.html_url,
+    });
+    addRepoDetailsInfo(table, 'Description :', repo.description);
+    addRepoDetailsInfo(table, 'Forks :', repo.forks);
+    addRepoDetailsInfo(table, 'Updated :', formatDate(repo.updated_at));
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repos) => {
-      const root = document.getElementById('root');
-      if (err) {
+  function renderContDetails(contributors, contUl) {
+    createAndAppend('h5', contUl, {
+      text: 'Contributions',
+      class: 'cont-title',
+    });
+    return contributors.forEach(contributor => {
+      const li = createAndAppend('li', contUl, { class: 'contribution' });
+      createAndAppend('img', li, {
+        src: contributor.avatar_url,
+        alt: `${contributor.login}'s image`,
+        class: 'img',
+      });
+      createAndAppend('a', li, {
+        href: contributor.html_url,
+        text: contributor.login,
+        target: '_blank',
+        class: 'link',
+      });
+      createAndAppend('span', li, {
+        text: contributor.contributions,
+        class: 'span-elm',
+      });
+    });
+  }
+  function renderContributors(selectedRepo, contUl) {
+    fetchJSON(selectedRepo.contributors_url)
+      .then(contributors => {
+        contUl.innerHTML = '';
+
+        renderContDetails(contributors, contUl);
+      })
+      .catch((err, root) => {
         createAndAppend('div', root, {
           text: err.message,
           class: 'alert-error',
         });
-        return;
-      }
-      const ul = createAndAppend('ul', root);
-      repos.forEach(repo => renderRepoDetails(repo, ul));
+      });
+  }
+  function renderPage(event, reposUl, contUl) {
+    const url = HYF_REPOS_URL;
+    fetchJSON(url)
```

Comment:
There is no need to fetch the repository information each time the user changes the selected repo. You have fetched it already in function main. It would be wasteful, network-wise, to fetch it again.